lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,119 @@ def post(self, request, *args, **kwargs):
+            fields.
+
+        id: (str) The Course's id (Course Run key)
+        name: (str) The course's name
+        number: (str) The course's number
+        org: (str) The course's organisation
+        start: (str) Date the course begins, in ISO 8601 notation
+        start_display: (str) Readably formatted start of the course
+        start_type: (str) Hint describing how `start_display` is set. One of:
+            * `"string"`: manually set by the course author
+            * `"timestamp"`: generated from the `start` timestamp
+            * `"empty"`: no start date is specified
+        end: (str) Date the course ends, in ISO 8601 notation
+        media: (dict) An object that contains named media items. Included here:
+            * course_image: An image to show for the course.  Represented
+              as an object with the following fields:
+                * uri: The location of the image
+        certificate: (dict) Information about the user's earned certificate in the course.
+            Included here:
+                * uri: The location of the user's certificate
+        is_self_paced: (bool) Indicates if the course is self paced
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent or the username was not provided
+        * 401 unauthorized, the provided access token has expired and is no longer valid
+          for an authenticated request.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    def get_certificate(self, request, course_id):
+        """Returns the information about the user's certificate in the course."""
+        if request.user.is_authenticated:
+            certificate_info = certificate_downloadable_status(request.user, course_id)
+            if certificate_info['is_downloadable']:
+                return {
+                    'url': request.build_absolute_uri(
+                        certificate_info['download_url']
+                    ),
+                }
+        return {}
+
+    # pylint: disable=arguments-differ
+    def list(self, request, **kwargs):
+        """
+        REST API endpoint for listing all the blocks information in the course and
+        information about the course while regarding user access and roles.
+
+        Arguments:
+            request - Django request object
+        """
+
+        response = super().list(request, kwargs)
+
+        if request.GET.get('return_type', 'dict') == 'dict':
+            course_id = request.query_params.get('course_id', None)
+            course_key = CourseKey.from_string(course_id)
+            course_overview = CourseOverview.get_from_id(course_key)
+
+            course_data = {
+                # identifiers
+                'id': course_id,
+                'name': course_overview.display_name,
```

Comment:
Can we move course overview data in another function and update dict here?
This will clean this view method.